pkg/reconciliation/reconcile_racks.go

```diff
@@ -2243,6 +2245,65 @@ func (rc *ReconciliationContext) fixMissingPVC() (bool, error) {
 	return false, nil
 }
 
+func (rc *ReconciliationContext) SetFullQueryLogging() result.ReconcileResult {
+	shouldFQLBeEnabled := false
+	dc := rc.GetDatacenter()
+	if dc.Spec.Config != nil {
+		var dcConfig map[string]interface{}
+		if err := json.Unmarshal(dc.Spec.Config, &dcConfig); err != nil {
+			rc.ReqLogger.Error(err, "error unmarshalling DC config JSON")
+			return result.RequeueSoon(2)
+		}
+		casYaml, found := dcConfig["cassandra-yaml"]
+		if !found {
+			return result.Continue()
+		}
+		casYamlMap, ok := casYaml.(map[string]interface{})
+		if !ok {
+			err := fmt.Errorf("type casting error")
+			rc.ReqLogger.Error(err, "couldn't cast cassandra-yaml value from config to map[string]interface{}")
+			return result.RequeueSoon(2)
+		}
+		if _, found := casYamlMap["full_query_logging_options"]; found {
+			serverMajorVersion, err := strconv.ParseInt(strings.Split(dc.Spec.ServerVersion, ".")[0], 10, 8)
+			if err != nil {
+				rc.ReqLogger.Error(err, "error parsing server major version. Can't enable full query logging without knowing this")
+				return result.RequeueSoon(2)
+			}
+			if serverMajorVersion < 4 {
+				err := fmt.Errorf("full query logging only supported on OSS Cassandra 4x+")
+				rc.ReqLogger.Error(err, "full_query_logging_options is defined in Cassandra config, it is not supported on the version of Cassandra you are running")
+				return result.RequeueSoon(2)
+			}
+			rc.ReqLogger.Info("full_query_logging_options is defined in Cassandra config, we will try to enable it via the management API")
+			shouldFQLBeEnabled = true
+		}
+	}
+
+	podList, err := rc.listPods(rc.Datacenter.GetClusterLabels())
+	if err != nil {
+		rc.ReqLogger.Error(err, "error listing all pods in the cluster to progress full query logging reconciliation")
+		return result.RequeueSoon(2)
+	}
+	for _, podPtr := range PodPtrsFromPodList(podList) {
+		fqlEnabledForPod, err := rc.NodeMgmtClient.CallIsFullQueryLogEnabledEndpoint(podPtr)
+		if err != nil {
+			rc.ReqLogger.Error(err, "can't get whether query logging enabled for pod ", "podName", podPtr.Name)
+			return result.RequeueSoon(2)
+		}
+		rc.ReqLogger.Info("full query logging status:", "isEnabled", fqlEnabledForPod, "shouldBeEnabled", shouldFQLBeEnabled)
+		if fqlEnabledForPod != shouldFQLBeEnabled {
+			rc.ReqLogger.Info("Setting full query logging on ", "podIP", podPtr.Status.HostIP, "podName", podPtr.Name, "fqlDesiredState", shouldFQLBeEnabled)
```

Comment:
Should `podPtr.Status.HostIP` be changed to `podPtr.Status.PodIP`?

Reply:
Is hostIP actually the IP of the underlying k8s node? I assumes it was the podIP but perhaps I'm wrong.

That does make me wonder why the CR even knows about the underlying node IPs?